o2a/utils/config_extractors.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Extractors for configuration and job-xml nodes"""
+from os import path
+from typing import Dict, List
+import xml.etree.ElementTree as ET
+
+from o2a.converter.constants import HDFS_FOLDER
+from o2a.converter.exceptions import ParseException
+from o2a.o2a_libs.property_utils import PropertySet
+from o2a.utils import el_utils
+
+TAG_CONFIGURATION = "configuration"
+TAG_PROPERTY = "property"
+TAG_NAME = "name"
+TAG_VALUE = "value"
+TAG_JOB_XML = "job-xml"
+
+
+def extract_properties_from_configuration_node(config_node: ET.Element, props: PropertySet) -> Dict[str, str]:
+    """Extracts configuration properties from ``configuration`` node"""
+    properties_dict: Dict[str, str] = dict()
+    for property_node in config_node.findall(TAG_PROPERTY):
+        name_node = property_node.find(TAG_NAME)
+        value_node = property_node.find(TAG_VALUE)
+
+        if name_node is None or value_node is None:
+            raise ParseException('Element "property" must have direct children elements: name, value')
```

Comment:
Won't this get caught already by the XSD validation?

Reply:
Yes, but Pylint does not know it.